File: src/leetcode/number-of-strings-that-appear-as-substrings-in-word/number-of-strings-that-appear-as-substrings-in-word.js
/**
 * Leetcode #1967
 * Difficulty: Easy
 * https://leetcode.com/problems/number-of-strings-that-appear-as-substrings-in-word/
 */
/**
 * @param {string[]} patterns
 * @param {string} word
 * @return {number}
 */
export const numOfStrings = (patterns, word) => {
  let ans = 0;

  for (const pattern of patterns) {
    if (word.includes(pattern)) {
      ans += 1;
    }
  }

  return ans;
};

File: src/leetcode/number-of-strings-that-appear-as-substrings-in-word/number-of-strings-that-appear-as-substrings-in-word.rs
impl Solution {
    pub fn num_of_strings(patterns: Vec<String>, word: String) -> i32 {
        patterns.iter().filter(|&pattern| word.contains(pattern)).count() as _
    }
}

// impl Solution {
//     pub fn num_of_strings(patterns: Vec<String>, word: String) -> i32 {
//         let mut ans = 0;
//
//         for pattern in patterns {
//             if word.contains(&pattern) {
//                 ans += 1;
//             }
//         }
//
//         ans
//     }
// }